src/nbl/asset/utils/CGLSLCompiler.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#include "nbl/asset/utils/CGLSLCompiler.h"
+#include "nbl/asset/utils/shadercUtils.h"
+
+#include <sstream>
+#include <regex>
+#include <iterator>
+
+
+using namespace nbl;
+using namespace nbl::asset;
+
+
+CGLSLCompiler::CGLSLCompiler(core::smart_refctd_ptr<system::ISystem>&& system)
+    : IShaderCompiler(std::move(system))
+{
+}
+
+core::smart_refctd_ptr<ICPUBuffer> CGLSLCompiler::compileToSPIRV(const char* code, const CGLSLCompiler::SOptions& options) const
+{
+    core::smart_refctd_ptr<ICPUBuffer> outSpirv = nullptr;
+    if (code)
+    {
+        core::smart_refctd_ptr<asset::ICPUShader> cpuShader;
+        if (options.includeFinder != nullptr)
+        {
+            cpuShader = resolveIncludeDirectives(code, options.stage, options.sourceIdentifier, options.maxSelfInclusionCount, options.logger);
+            if (cpuShader)
+            {
+                code = reinterpret_cast<const char*>(cpuShader->getContent()->getPointer());
+            }
+        }
+
+        if (strcmp(options.entryPoint, "main") == 0)
+        {
+            shaderc::Compiler comp;
+            shaderc::CompileOptions shadercOptions; //default options
+            shadercOptions.SetTargetSpirv(static_cast<shaderc_spirv_version>(options.targetSpirvVersion));
+            const shaderc_shader_kind stage = options.stage == IShader::ESS_UNKNOWN ? shaderc_glsl_infer_from_source : ESStoShadercEnum(options.stage);
+            const size_t glsl_len = strlen(code);
+            if (options.genDebugInfo)
+                shadercOptions.SetGenerateDebugInfo();
+
+            shaderc::AssemblyCompilationResult asm_res;
+            shaderc::SpvCompilationResult bin_res;
+            if (options.outAssembly)
+            {
+                asm_res = comp.CompileGlslToSpvAssembly(code, glsl_len, stage, options.sourceIdentifier ? options.sourceIdentifier : "", shadercOptions);
+                options.outAssembly->resize(std::distance(asm_res.cbegin(), asm_res.cend()));
+                memcpy(options.outAssembly->data(), asm_res.cbegin(), options.outAssembly->size());
+                bin_res = comp.AssembleToSpv(options.outAssembly->data(), options.outAssembly->size(), shadercOptions);
+            }
```

Comment:
@Crisspl why did we need to keep the assembly around?

Reply:
I can't find any other reference to `outAssembly` outside of `IGLSLCompiler` even in the examples.
NUKE IT?!

Reply:
![image](https://user-images.githubusercontent.com/6894321/203713413-d7c97f1b-b751-4f7a-b5a5-131e15294133.png)

Yes but actually do it, don't do blabbing on about it unlike some other people.